README.md

```diff
@@ -89,32 +126,42 @@ Make sure your `AndroidManifest.xml` includes intent filters:
     </activity>
 ```
 
-Then, use the `AuthReceiver` in your main activity:
+Then, use the `BroadcastReceiver` in your main activity:
+
+The `BroadcastReceiver` listens for authentication results and extracts the authorization code after a sign-in attempt, enabling the authentication flow in your app.
+
 ```kotlin
-    var receivedAuthorizationCode by remember { mutableStateOf<String?>(null) }
-    val filter = IntentFilter("io.projectliberty.helpers.AUTH_RESULT")
-    val receiver = object : BroadcastReceiver() {
-        override fun onReceive(context: Context?, intent: Intent?) {
-            receivedAuthorizationCode = intent?.getStringExtra("authorizationCode")
-            Log.d("HostApp", "✅ Received Auth Code: $receivedAuthorizationCode")
+    setContent {
+        var authorizationCode by remember { mutableStateOf<String?>(null) }
+    
+        val authReceiver = remember {
+            object : BroadcastReceiver() {
+                override fun onReceive(context: Context?, intent: Intent?) {
+                    val receivedCode = intent?.getStringExtra(AuthConstants.AUTH_INTENT_KEY)
+                    authorizationCode = receivedCode
+                }
+            }
         }
+    
+        val authFilter = IntentFilter(AuthConstants.AUTH_RESULT_ACTION)
+    
+        ContextCompat.registerReceiver(
+            this,
+            authReceiver,
+            authFilter,
+            ContextCompat.RECEIVER_NOT_EXPORTED
+        )
+    //    ...
```

Comment:
my brain can't think of what do do here to say that there would be more stuff here

Reply:
I added a few comments elsewhere on how to lead them to the next steps. I don't think there is anything more here specifically